File: 121/solution.py
class Solution(object):
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        
        if len(prices) == 1:
            return 0
        minPrice = prices[0]
        profit = 0
        
        for i in prices:
            minPrice = min(minPrice, i)
            profit = max(profit, i-minPrice)
        return profit
        
            
            

File: 121/solution.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int len = prices.size();
        if(len==1) return 0;
        
        int minprice = prices[0];
        int profit = 0;
        for(int i=0; i<len; ++i){
            minprice = min(minprice, prices[i]);
            profit = max(prices[i]-minprice, profit);
        }
        return profit;
    }
};
